pyccel/parser/grammar/openmp.tx

```diff
@@ -1,184 +1,102 @@
-OmpFirstPrivate: 'firstprivate' '(' args+=ID[','] ')';
-OmpLastPrivate: 'lastprivate' '(' args+=ID[','] ')';
-OmpCopyin: 'copyin' '(' args+=ID[','] ')';
-OmpDepend: 'depend' '('dtype=OmpDependenceType ':' args+=ID[','] ')';
-OmpReduction: 'reduction' '('op=OmpReductionOperator ':' args+=ID[','] ')';
-OmpinReduction: ctype='in_reduction' '('op=OmpReductionOperator ':' args+=ID[','] ')';
-OmpCollapse: 'collapse' '(' n=IntegerType ')';
-OmpLinear: 'linear' '(' val=ID ':' step=IntegerType ')';
-OmpOrdered: 'ordered' ('(' n=IntegerType ')')?;
-OmpSchedule: 'schedule' '(' kind=OmpScheduleKind (',' chunk_size=IntegerType)? ')';
-OmpEndClause: 'end' construct*=OpenmpConstructs;
-OmpCriticalName: '(' args=ID ')';
-OmpFilter: name='filter' '(' n=IntegerType ')';
-OmpPriority: name='priority' '(' n=PriorityIndex ')';
-OmpUntied: name='untied';
-OmpMergeable: name='mergeable';
-OmpNogroup: name='nogroup';
-OmpAtomicClause: name=AtomicClause;
-AtomicMemoryClause: name=AtomicMClause;
-FlushList: '(' args+=ID[','] ')';
-OmpCancelType: name=CancelType;
-OmpNowait: name='nowait';
+OmpEndedConstruct: 'parallel';
+ParenthesizedExpression: (('(' ParenthesizedExpression ')') | /[^()]*/)*; // TODO: find a better way to do this as this will not escape the parenthesis inside of a string
+DataSharingAttribute: ('shared' | 'none' | 'private' | 'firstprivate');
+ReductionOperator: ('+' | '-' | '*' | '&' | '|' | '^' | '&&' | '||' | 'min' | 'max'); // TODO: those are only the implicitly declared ones for C and C++. Add the others for Fortran and support user-defined ones (see specs https://www.openmp.org/spec-html/5.0/openmpsu107.html#x140-581002)
+AffinityPolicy: ('close' | 'spread');
+ReductionModifier: ('inscan' | 'task' | 'default');
+
 ////////////////////////////////////////////////////
 
 ////////////////////////////////////////////////////
-IntegerType: ( INT | ID );
-CancelType: ('parallel' | 'sections' | 'taskgroup' | 'for');
-OmpScheduleKind: ('static' | 'dynamic' | 'guided' | 'auto' | 'runtime' );
-OmpProcBindStatus: ('master' | 'close' | 'spread');
-OmpReductionOperator: ('+' | '-' | '*' | '/');
-OmpDependenceType: ('in' | 'out' | 'inout' | 'mutexinoutset' | 'depobj');
-OmpDefaultStatus: ('private' | 'firstprivate' | 'shared' | 'none');
-OmpMapType: (('always' | 'close' | ('mapper' '(' ID ')') ',')? ('tofrom' | 'from' | 'to' | 'alloc' | 'release' | 'delete')) ':';
-OpenmpConstructs: ('simd' | 'masked' | 'master' | 'single' | 'parallel' | 'for' | 'critical' ('(' ID  ')')? | 'task' | 'target' | 'teams' | 'sections' | 'section');
-
-AtomicClause: ('write' | 'read' | 'update' | 'capture' | 'compare' | 'weak');
-AtomicMClause: ('seq_cst' | 'acq_rel' | 'release' | 'acquire' | 'relaxed');
-ThreadIndex: (ID | INT);
-TaskIndex: (ID | INT);
-TeamIndex: (ID | INT);
-PriorityIndex: (ID | INT);
-NotaStmt: /.*$/;
+//              Omp Versions
 ////////////////////////////////////////////////////
+OMP_4_5: '';
+OMP_5_0: '';
+OMP_5_1: '';
```

Comment:
Am I right in thinking that any new OpenMP versions will lead to big `textx` errors until the new constructs are implemented?

Reply:
I don't see where the problem will be. When a new version is out we don't need to change anything if we don't want to support that version yet. if the user did use syntax from the new version he will get an error that the thing he uses is unknown the same as if he uses a wrong syntax and that's it

Reply:
No, textx errors are now caught https://github.com/pyccel/pyccel/blob/rewrite-omp/pyccel/parser/syntax/openmp.py#L59